pkg/resourcesets/collector.go

```diff
@@ -248,157 +174,147 @@ func (h *ResourceHandler) filterByLabel(ctx context.Context, dr dynamic.Resource
 	return resourceObjectsList, nil
 }
 
+func (h *ResourceHandler) filterByKind(filter v1.ResourceSelector, apiResources []k8sv1.APIResource) ([]k8sv1.APIResource, error) {
+	var resourceList []k8sv1.APIResource
+	var kindRegexp *regexp.Regexp
+	if filter.KindsRegexp == "" && len(filter.Kinds) == 0 {
+		// if no filters for resource kind are given, return entire resource list
+		return apiResources, nil
+	}
+	if filter.KindsRegexp != "" {
+		var err error
+		kindRegexp, err = regexp.Compile(filter.KindsRegexp)
+		if err != nil {
+			return resourceList, err
```

Comment:
(unsure): In this scenario we're returning an empty slice and an `err`. I haven't tested this to confirm myself, but from an end-user POV we def want this type of error to cause the backup to fail completely. And ideally provide logging/error messages that will direct us to the (in this case) filter with Regex that caused errors.

In other words, if we have to chose between:
1. An error causing an empty backup to be taken and the backup marked completed, or
2. A loud noisy error and failed backup.
We certainly want to see option 2 be the outcome.

Reply:
(this shouldn't be a blocker of this PR per say, just a general curiosity I have that I haven't tested in a lab to check the results yet)

Reply:
I kept this from the previous code, but I agree with you: if we're returning an non-nil error, the first return value should usually be nil (because the caller should generally not have to deal with a partially correct result from a function that errors out).